### PR TITLE
feat: add support for folder drag 'n' drop

### PR DESCRIPTION
**Pull request by rolandjitsu**
**What kind of change does this PR introduce?**

- [ ] bugfix
- [x] feature
- [ ] refactoring / style
- [ ] build / chore
- [ ] documentation

**Did you add tests for your changes?**

- [x] Yes, my code is well tested
- [ ] Not relevant

**If relevant, did you update the documentation?**

- [x] Yes, I've updated the documentation
- [ ] Not relevant

**Summary**

Add support for folder drag 'n' drop by default, no need to install a different library. It should have no effect on users that use their own impl. for file aggregation.

**Does this PR introduce a breaking change?**

No.

**Other information**

### CODE REVIEW

File: src/index.spec.js [hunk 1]
```diff
@@ -1202,6 +1208,7 @@ describe('Dropzone', () => {
         </Dropzone>
       )
       await dropzone.simulate('drop', createDtWithFiles(files))
+      await flushPromises(dropzone)
```
Comment (okonet) [inline src/index.spec.js]: Question: why is this needed and can’t we put it into `afterEach` for example?

Comment (rolandjitsu) [reply src/index.spec.js]: I think it was introduced by you or someone else. Without it, tests will fail. It's probably because of the file aggregation which happens async.

Comment (okonet) [reply src/index.spec.js]: Yes it’s needed because of async get items. But the question still stands: can we remove the repetition and put it into `afterEach`?

Comment (rolandjitsu) [reply src/index.spec.js]: Will try it and update if it does.

Comment (rolandjitsu) [reply src/index.spec.js]: Nope, it does not work if I move `flushPromises()` to `afterEach`.

Comment (rolandjitsu) [reply src/index.spec.js]: @okonet so, what do you think, should we merge this?

**Comment by okonet** [comment]
LGTM please merge and release as a breaking change.

**Comment by rolandjitsu** [comment]
Hmm, it did not release as breaking change. Maybe it's because I did not add the colons after `BREAKING CHANGE`?

**Comment by okonet** [comment]
Oh I’m so sick of how strict and fragile the semantic-release is :(

**Comment by rolandjitsu** [comment]
Should be fixed with https://github.com/react-dropzone/react-dropzone/pull/767.
